java/code/src/com/redhat/rhn/frontend/action/systems/sdc/UpdateCustomDataAction.java

```diff
@@ -134,6 +135,9 @@ public ActionForward execute(ActionMapping mapping,
             cdv.setCreator(user);
             cdv.setLastModifier(user);
             request.setAttribute(VAL_PARAM, form.get(VAL_PARAM));
+            server.asMinionServer().ifPresent(minion -> {
+                MinionPillarManager.INSTANCE.generatePillar(minion);
+            });
```

Comment:
I am a bit confused by the logic from lines above this. It seems to me that `addCustomDataValue` already creates a new `CustomDataValue` instance and sets it if null, and now we have the logic repeated here.

I would not mind in the context of this PR normally, but here we are adding code which is clearly back-end (pillar generation) to a front-end class (an Action), and was wondering where it would be placed best.

Would it be possible to fold the code above and the pillar generation inside of `addCustomDataValue` - either to be left in `Server` or moved into `ServerFactory`?